### ISSUE TITLE
LSP should filter duplicate results

### ISSUE THREAD

**Issue by dev-ardi** [helix-editor/helix#6148]
### Summary

When I go to references (`gr`) helix shows this:
![image](https://user-images.githubusercontent.com/47633543/222081934-496dbda8-3202-4a0d-9548-0e4da9c5b0ae.png)

I assume this is a problem in rust-analyzer but regardless we should filter duplicates in the client.

### Reproduction Steps

 

I tried this:

1. `hx`

I expected this to happen:

Instead, this happened:

### Helix log

Helix.log didn't have any relevant information.

### Platform

Windows

### Terminal Emulator

WindowsTerminal_1.16.10261.0_x64

### Helix Version

helix 22.08 (ee94031f)

**Comment by askreet** [comment]
I agree this is almost certainly a bug in rust-analyzer. Any chance this code is public where I could reproduce it?

I don't think filtering the results from the LSP is a good idea, in general.
